src/core/api/clock.ts

```diff
@@ -336,12 +336,12 @@ function createClock(
 
     function getCurrentClockTick(state : IMediaInfosState) : IClockTick {
       const mediaTimings = getMediaInfos(mediaElement, state);
+      log.debug("API: current media element state", mediaTimings);
       const stalledState = getStalledStatus(lastTimings, mediaTimings, options);
 
-      // /!\ Mutate mediaTimings
-      return objectAssign(mediaTimings,
-                          { stalled: stalledState,
-                            getCurrentTime: () => mediaElement.currentTime });
+      return objectAssign({ stalled: stalledState,
+                            getCurrentTime: () => mediaElement.currentTime },
+                          mediaTimings);
```

Comment:
By doing reverse order, we risk to override `stalled` and `getCurrentTime` in the future.

If the need was to avoid mutating `mediaTimings`, maybe add an empty object as first argument?